processor/src/operations/crypto_ops.rs

```diff
@@ -87,6 +87,13 @@ where
         // Asserting the computed root of the Merkle path from the advice provider is consistent with
         // the input root.
         assert_eq!(root, computed_root, "inconsistent Merkle tree root");
+        if root == computed_root {
```

Comment:
We should probably remove `assert_eq` from here and the the condition for the `if` statement should be `root != computed_root` - right?

Reply:
Also, the comment should be updated.

Reply:
Oops - and it should have been `!=`. Fixed